applicationset/services/scm_provider/gitlab_test.go

```diff
@@ -19,7 +19,7 @@ func gitlabMockHandler(t *testing.T) func(http.ResponseWriter, *http.Request) {
 		switch r.RequestURI {
 		case "/api/v4":
 			fmt.Println("here1")
-		case "/api/v4/groups/test-argocd-proton/projects?include_subgroups=false&per_page=100":
+		case "/api/v4/groups/test-argocd-proton/projects?include_subgroups=false&per_page=100", "/api/v4/groups/test-argocd-proton/projects?include_subgroups=false&per_page=100&with_shared=false", "/api/v4/groups/test-argocd-proton/projects?include_subgroups=true&per_page=100&with_shared=false", "/api/v4/groups/test-argocd-proton/projects?include_subgroups=true&per_page=100&with_shared=true":
```

Comment:
Could this mock be modified to return different results for `with_shared` true vs. false, so that the test can confirm that the behavior is in fact different for those two configurations? It would basically just confirm that the variable is in fact being piped all the way to the request param.

Reply:
maybe... will have to debug how gitlab manages shared projects... having a look

Reply:
I updated the test so we check the `shared project` specificity... please review. Thanks

Reply:
OK better now... again i'm not a testing expert... this sounds hacky... but this whole test is a hack as we hardcode the gitlab server expected output.
In this sense we're not really testing the subgroups flag... but it wasn't tested before either...

Reply:
I also split the test for the `subgroups` only... 
I don't think we have to test both options at the same time ? 

TODO: create a shared-project and a subgroup project in the public gitlab repo at https://gitlab.com/test-argocd-proton